presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/SelectionContext.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.resourceGroups;
+
+public class SelectionContext<C>
+{
+    private final ResourceGroupId resourceGroupId;
+    private final C context;
+
+    public SelectionContext(ResourceGroupId resourceGroupId, C context)
+    {
+        this.resourceGroupId = resourceGroupId;
+        this.context = context;
```

Comment:
Probably best to make this non-null as well. Can use a fake enum instead of void:
```java
enum VoidContext { NONE }